python/cudf/cudf/tests/test_scalar.py

```diff
@@ -1,512 +1,11 @@
-            marks=pytest.mark.skipif(
-                pa._generated_version.version_tuple[0] < 19,
-                reason="decimal64 format string only supported in pyarrow>=19",
-            ),
-        ),
-        cudf.Decimal128Dtype,
-    ],
-)
-def test_device_scalar_direct_construction(value, decimal_type):
-    value = cudf.utils.dtypes.to_cudf_compatible_scalar(value)
-
-    dtype = (
-        value.dtype
-        if not isinstance(value, Decimal)
-        else decimal_type._from_decimal(value)
-    )
-
-    s = cudf.Scalar(value, dtype)
-
-    assert s.value == value or np.isnan(s.value) and np.isnan(value)
-    if isinstance(
-        dtype, (cudf.Decimal64Dtype, cudf.Decimal128Dtype, cudf.Decimal32Dtype)
-    ):
-        assert s.dtype.precision == dtype.precision
-        assert s.dtype.scale == dtype.scale
-    elif dtype.char == "U":
-        assert s.dtype == "object"
-    else:
-        assert s.dtype == dtype
-
-
-@pytest.mark.parametrize("value", SCALAR_VALUES + DECIMAL_VALUES)
-def test_construct_from_scalar(value):
-    value = cudf.utils.dtypes.to_cudf_compatible_scalar(value)
-    x = cudf.Scalar(
-        value, value.dtype if not isinstance(value, Decimal) else None
-    )
-    y = cudf.Scalar(x)
-    assert x.value == y.value or np.isnan(x.value) and np.isnan(y.value)
-
-    # check that this works:
-    y.device_value
-
-    x._is_host_value_current == y._is_host_value_current
-    x._is_device_value_current == y._is_device_value_current
-
-
-@pytest.mark.parametrize(
-    "data", ["20000101", "2000-01-01", "2000-01-01T00:00:00.000000000", "2000"]
-)
-@pytest.mark.parametrize("dtype", DATETIME_TYPES)
-def test_datetime_scalar_from_string(data, dtype):
-    slr = cudf.Scalar(data, dtype)
-
-    expected = np.datetime64(datetime.datetime(2000, 1, 1)).astype(dtype)
-
-    assert expected == slr.value
-
-
-def test_scalar_cache():
-    s = cudf.Scalar(1)
-    s2 = cudf.Scalar(1)
-
-    assert s is s2
-
-
-def test_scalar_cache_rmm_hook():
-    # test that reinitializing rmm clears the cuDF scalar cache, as we
-    # register a hook with RMM that does that on reinitialization
-    s = cudf.Scalar(1)
-    s2 = cudf.Scalar(1)
-
-    assert s is s2
-
-    rmm.reinitialize()
-
-    s3 = cudf.Scalar(1)
-    assert s3 is not s
-
-
-def test_default_integer_bitwidth_scalar(default_integer_bitwidth):
-    # Test that integer scalars are default to 32 bits under user options.
-    slr = cudf.Scalar(128)
-    assert slr.dtype == np.dtype(f"i{default_integer_bitwidth // 8}")
-
-
-def test_default_float_bitwidth_scalar(default_float_bitwidth):
-    # Test that float scalars are default to 32 bits under user options.
-    slr = cudf.Scalar(128.0)
-    assert slr.dtype == np.dtype(f"f{default_float_bitwidth // 8}")
-
-
-def test_scalar_numpy_casting():
-    # binop should upcast to wider type
-    s1 = cudf.Scalar(1, dtype=np.int32)
-    s2 = np.int64(2)
-    assert s1 < s2
 
 
 def test_construct_timezone_scalar_error():
```

Comment:
any reason we're keeping just this one test?

Reply:
We are still using the `to_cudf_compatible_scalar` utility function internally in places so I thought it was worth keeping. I'm hoping to remove it and this test in a future pass.